spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,649 @@
+                .forEach(entry -> {
+                    try {
+                        productCount.incrementAndGet();
+                        String name = new File(entry.getName()).getName().replace(".json", "");
+                        ProductDTO expected = productIdToProduct.get(name);
+                        assertNotNull(expected);
+                        ProductDTO actual = ExportUtil
+                            .deserializeJsonFile(export, entry.getName(), ProductDTO.class);
+                        assertEquals(expected.getId(), actual.getId());
+                        assertEquals(expected.getName(), actual.getName());
+                    }
+                    catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+
+            assertEquals(productIdToProduct.size(), productCount.get());
+        }
+
+        @Test
+        @OnlyInStandalone
+        @DisplayName("should export products and certs")
+        void shouldExportProductsAndCerts() throws Exception {
+            int target = PRODUCTS_PATH.length() - 1;
+            AtomicInteger productCount = new AtomicInteger(0);
+            AtomicInteger expectedNumberOfCerts = new AtomicInteger(0);
+            AtomicInteger actualNumberOfCerts = new AtomicInteger(0);
+            export.stream()
+                .filter(entry -> entry.getName().startsWith(PRODUCTS_PATH))
+                .filter(entry -> entry.getName().lastIndexOf('/') == target)
+                .forEach(entry -> {
+                    if (entry.getName().endsWith(".json")) {
+                        productCount.incrementAndGet();
+                        String fileName = new File(entry.getName()).getName().replace(".json", "");
+                        ProductDTO product = productIdToProduct.get(fileName);
+                        assertNotNull(product);
+                        // Count numeric ids
+                        try {
+                            Integer.parseInt(fileName);
+                            expectedNumberOfCerts.incrementAndGet();
+                        }
+                        catch (NumberFormatException e) {
+                            // ExpectedName is not a numeric id
+                        }
+                    }
+
+                    if (entry.getName().endsWith(".pem")) {
+                        actualNumberOfCerts.incrementAndGet();
+                    }
+                });
+
+            assertEquals(productIdToProduct.size(), productCount.get());
+            assertEquals(expectedNumberOfCerts.get(), actualNumberOfCerts.get());
+        }
+
+        @Test
+        @DisplayName("should not schedule non cron tasks")
+        public void shouldNotScheduleNonCronTasks() throws Exception {
+            assertForbidden(() -> jobsApi.scheduleJob(UNDO_IMPORTS_JOB_KEY));
+        }
+
+        private void initializeData() throws Exception {
```

Comment:
@Ceiu @JoshAlbrecht23 Even though I understand we don't need an 'exporter' abstraction in the same way we had in the ruby tests, we still have about 6 `import_*.rb` spec tests to be migrated that need this kind of data created in a manifest for importing. Can we make this generic and expose it for reuse? It's just too large for copy and pasting

Reply:
We could have an ExportFactory that has a builder that lets you define what should be included in the export. Then have the ExportFactory be able to return the objects that were created so that you can assert against the export. That way you can use the ExportFactory in both the ExportSpecTests and the ImportSpecTests and can easily create an export based on just what you need in each individual test.

```
ExportFactory factory = new ExportFactory.Builder()
    .withVirtualProductPool()
    .withDerivedProduct()
    .build();

ZipFile export = factory.createExport();

```

Reply:
While it could work, this might be a bit too overengineered for what we need. I agree with @Ceiu now that we should probably let this be as is, and the import spec tests will define their own data, because we really don't need everything from this initializeData method. Especially since most of those import_*.rb files have only 1-2 tests each, so they need minimal amounts of data in the manifest, and adding an abstraction layer would just be too much. So, tl;dr, disregard my comment!